app/src/components/Modals/add-activity-modal.tsx

```diff
@@ -139,10 +143,59 @@ const AddActivityModal: FC<AddActivityModalProps> = ({
   const [createActivityValue, { isLoading }] =
     api.useCreateActivityValueMutation();
 
-  const onSubmit: SubmitHandler<Inputs> = async (data) => {
-    setHasActivityData(!hasActivityData);
-    await createActivityValue({ inventoryId, data }).then((res: any) => {
+  function extractGasesAndUnits(data: any) {
+    const gases = ["CH4", "CO2", "N2O"];
+    const gasArray: { gas: string; factor: number; unit: string }[] = [];
+    gases.forEach((gas) => {
+      const gasFactorKey = `${gas}EmissionFactor`;
+      const gasUnitKey = `${gas}EmissionFactorUnit`;
+      const gasObject = {
+        gas: gas,
+        factor: data[gasFactorKey],
+        unit: data[gasUnitKey],
+      };
+
+      gasArray.push(gasObject);
+    });
+    return gasArray;
+  }
+
+  const onSubmit: SubmitHandler<Inputs> = async ({ activity }) => {
+    const gasValues = extractGasesAndUnits(activity);
+    const requestData = {
+      activityData: {
+        co2_amount: gasValues[1].factor,
+        ch4_amount: gasValues[0].factor,
+        n2o_amount: gasValues[2].factor,
+        activityType: activity.buildingType,
+        fuelType: activity.fuelType,
```

Comment:
This case inconsistency is a bit annoying. Let's figure out a standard casing (e.g. snake_case) and stick with it, so we don't have to guess for each input how it was formatted.
So `activity_type` and `fuel_type` here. Or using the keys from the hierarchy JSON, which use `kebab-case`.